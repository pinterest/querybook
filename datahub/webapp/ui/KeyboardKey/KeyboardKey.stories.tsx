import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { KeyboardKey } from './KeyboardKey';

export const _KeyboardKey = () => (
    <div>
        <KeyboardKey value="esc" />
        <div className="mt16">
            <KeyboardKey value="âŒ˜" />
            <span className="pr4">+</span>
            <KeyboardKey value="K" className="mr4" />
        </div>
        <div className="mt16">
            <KeyboardKey value="Alt" />
            <span className="pr4">+</span>
            <KeyboardKey value="F4" className="mr4" />
        </div>
    </div>
);

export default {
    title: 'Stateless/KeyboardKey',
    decorators: [centered],
};
