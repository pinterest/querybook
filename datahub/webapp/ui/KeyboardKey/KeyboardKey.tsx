import styled from 'styled-components';
import { isOSX } from 'lib/utils/keyboard';
import React, { useMemo } from 'react';

const StyledKeyboardKey = styled.span.attrs({
    className: 'KeyboardKey mr4',
})`
    cursor: default;
    border: var(--border);
    border-radius: var(--border-radius);
    background-color: var(--bg-color);
    padding: 4px 12px;
    font-family: var(--family-monospace);
    font-size: var(--small-text-size);
    text-transform: lowercase;
`;

const OSXToWindows = {
    '⌘': 'ctrl',
    '⌥': 'alt',
};

export const KeyboardKey: React.FC<{
    className?: string;
    value: string;
}> = ({ value, className }) => {
    const mappedKey = useMemo(
        () => (!isOSX && value in OSXToWindows ? OSXToWindows[value] : value),
        [value]
    );
    return (
        <StyledKeyboardKey className={className}>{mappedKey}</StyledKeyboardKey>
    );
};
