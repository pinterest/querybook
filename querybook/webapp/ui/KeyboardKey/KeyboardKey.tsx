import React, { useMemo } from 'react';
import styled from 'styled-components';

import { getKeySymbol } from 'lib/utils/keyboard';

const StyledKeyboardKey = styled.span.attrs({
    className: 'KeyboardKey mr4',
})`
    cursor: default;
    border: var(--border);
    border-radius: var(--border-radius-sm);
    background-color: var(--bg-light);
    padding: 4px 12px;
    font-family: var(--font-monospace);
    font-size: var(--small-text-size);
    text-transform: lowercase;
`;

export const KeyboardKey: React.FC<{
    className?: string;
    value: string;
}> = ({ value, className }) => {
    const mappedKey = useMemo(() => getKeySymbol(value), [value]);
    return (
        <StyledKeyboardKey className={className}>{mappedKey}</StyledKeyboardKey>
    );
};
