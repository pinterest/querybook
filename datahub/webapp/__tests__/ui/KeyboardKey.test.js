import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { KeyboardKey } from '../../ui/KeyboardKey/KeyboardKey';

it('renders without crashing', () => {
    shallow(<KeyboardKey value="test" />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<KeyboardKey value="test" />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(<KeyboardKey value="test" />);
        expect(output).toMatchSnapshot();
    });
});
