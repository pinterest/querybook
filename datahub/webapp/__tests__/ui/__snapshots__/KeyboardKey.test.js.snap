// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<styled.span>
  test
</styled.span>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<span
  className="sc-bdfBwQ gEkjil KeyboardKey mr4"
>
  test
</span>
`;
